teps/0040-ignore-step-errors.md

```diff
@@ -145,13 +156,304 @@ This proposal is limited to a step within a task and does not address `pipelineT
+
+Introduce a new section `exit` as part of the
+[steps](https://github.com/tektoncd/pipeline/docs/tasks.md#defining-steps) definition along with an `image` and `script`.
+
+```
+- name: failing-step
+  image: alpine
+  exit:
+    onError: [ continue | fail]
+```
+
+The `Step` struct will have a new field `Exit`:
+
+```go
+type Step struct {
+
+    corev1.Container `json:",inline"`
+
+    // Script is the contents of an executable file to execute.
+    Script string `json:"script,omitempty"`
+
+    // ...
+
+    // define the exiting behavior of the container in this section
+    // takes an additional field `continue`
+    // set continue to afterSuccess (default) to indicate the entrypoint to exit the taskRun if the container exits with non zero exit code
+    // set continue to always to indicate the entrypoint to continue executing the rest of the steps irrespective of the container exit code
+    Exit *Exit `json:"exit,omitempty"`
+}
+```
+
+This new section `exit` is a `struct` type with one field:
+
+```go
+type Exit struct {
+	// can be set to [ continue | fail ]
+    // continue indicates continue executing the rest of the steps irrespective of the container exit code
+    // fail indicates exit the taskRun if the container exits with a non zero exit code
+    OnError string `json:"onError,omitempty"`
+}
+```
+
+A `task` author or a `pipeline` author can use `exit` section to ignore the step error. If `onError` is
+set to `continue`, the entrypoint sets the original failed exit code of the `script` or the wrapped command in the
+container termination state.
+
+To ignore a step error, set the `onError` to `continue`:
+
+```yaml
+steps:
+  - image: docker.io/library/golang:latest
+    name: ignore-unit-test-failure
+    exit:
+      onError: continue
+    script: |
+      go test .
+```
+
+A `step` with `onError` set to `continue` does not fail the `taskRun` and continues executing the rest of the steps in
+a task. The original failed exit code of the wrapped command is available in the termination state of the container.
+
+```
+kubectl get tr taskrun-unit-test-t6qcl -o json | jq .status
+{
+  "completionTime": "2021-06-21T18:22:06Z",
+  "conditions": [
+    {
+      "lastTransitionTime": "2021-06-21T18:22:06Z",
+      "message": "All Steps have completed executing",
+      "reason": "Succeeded",
+      "status": "True",
+      "type": "Succeeded"
+    }
+  ],
+  "podName": "taskrun-unit-test-t6qcl-pod-zpqs9",
+  "startTime": "2021-06-21T18:21:57Z",
+  "steps": [
+    {
+      "container": "step-ignore-unit-test-failure",
+      "imageID": "...",
+      "name": "ignore-unit-test-failure",
+      "terminated": {
+        "containerID": "...",
+        "exitCode": 1,
+        "finishedAt": "2021-06-21T18:22:05Z",
+        "reason": "Completed",
+        "startedAt": "2021-06-21T18:22:05Z"
+      }
+    },
+  ],
+```
+
+As part of this design, we are introducing an additional internal volume `/tekton/steps/`. A file named `exitCode` will
```

Comment:
i like the name of this volume - im wondering if it would make sense for the debug on breakpoint feature to use the same volume, or if we need a distinct volume for each? (https://github.com/tektoncd/community/blob/main/teps/0042-taskrun-breakpoint-on-failure.md#mounts) ( @waveywaves )